mobile-widgets/uploadDiveLogsDE.cpp

```diff
@@ -257,7 +257,18 @@ void uploadDiveLogsDE::uploadDives(const QString &filename, const QString &useri
 
 void uploadDiveLogsDE::updateProgress(qint64 current, qint64 total)
 {
-	qDebug() << "---> reply Progress " << current << "/" << total;
+	if (!reply)
+		return;
+	if (total <= 0 || current <= 0)
+		return;
+
+	// Calculate percentage
+	// And signal whoever wants to know
+	qreal percentage = (float)current / (float)total;
+	emit uploadProgress(percentage);
+
+	// reset the timer: 30 seconds after we last got any data
+	timeout.start();
```

Comment:
that's a **LONG** timeout

Reply:
Yup, copied from the desktop version. What would you like as timeout ?

Reply:
Ha. That's funny. It shows you just how inconsistent reviews here are. We have a command line option to increase the timeout when accessing the subsurface cloud.
Maybe we could reuse that here? `prefs.cloud_timeout`.
This is set to 5 seconds by default in the desktop app, but 10 seconds in the mobile app, because we felt that you'd be more likely to have bad internet on a device than on your laptop (not always true, though). And then there is the `--cloud-timeout=` command line option to overwrite this.